### PR TITLE
Add position() function to PPL

### PR DESCRIPTION
**Pull request by margarit-h**
Signed-off-by: Margarit Hakobyan <margarith@bitquilltech.com>

### Description
Usage: The syntax POSITION(substr IN str) returns the position of the first occurrence of substring substr in string str. Returns 0 if substr is not in str. Returns NULL if any argument is NULL.

Argument type: STRING, STRING

Return type integer:

(STRING IN STRING) -> INTEGER

Example::

    opensearchsql> source=location | eval `POSITION('world' IN 'helloworld')` = POSITION('world' IN 'helloworld'), `POSITION('invalid' IN 'helloworld')`= POSITION('invalid' IN 'helloworld')  | fields `POSITION('world' IN 'helloworld')`, `POSITION('invalid' IN 'helloworld')`;
    fetched rows / total rows = 2/2
    +-------------------------------------+---------------------------------------+
    | POSITION('world' IN 'helloworld')   | POSITION('invalid' IN 'helloworld')   |
    |-------------------------------------+---------------------------------------|
    | 6                                   | 0                                     |
    | 6                                   | 0                                     |
    +-------------------------------------+---------------------------------------+

 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — APPROVED** [review]

File: docs/user/dql/functions.rst [hunk 1]
```diff
@@ -219,7 +219,7 @@ INTEGER/LONG/FLOAT/DOUBLE -> DOUBLE
 
 Example::
 
-    opensearchsql> SELECT CBRT(8), CBRT(9.261), CBRT(-27);
+    os> SELECT CBRT(8), CBRT(9.261), CBRT(-27)
```
Comment (GumpacG) [inline docs/user/dql/functions.rst:222]: Are these related to position? If not, it may be worthwhile mentioning why it's included in the PR description.

File: docs/user/ppl/functions/string.rst [hunk 2]
```diff
@@ -150,6 +150,31 @@ Example::
...
 
+POSITION
+------
+
+Description
+>>>>>>>>>>>
+
+Usage: The syntax POSITION(substr IN str) returns the position of the first occurrence of substring substr in string str. Returns 0 if substr is not in str. Returns NULL if any argument is NULL.
+
+Argument type: STRING, STRING, INTEGER
```
Comment (GumpacG) [inline docs/user/ppl/functions/string.rst:161]: Does `POSITION` accept integers as an argument?

File: docs/user/dql/functions.rst (hunk 1 above)
Comment (margarit-h) [reply docs/user/dql/functions.rst:222]: These tests will run only when started with `os>` , otherwise they will be skipped. After Yuri mentioned that with regards to these position() related PRs, I identified  the above and got it fixed as soon as possible.

File: docs/user/ppl/functions/string.rst (hunk 2 above)
Comment (margarit-h) [reply docs/user/ppl/functions/string.rst:161]: No, the position() syntax is: `POSITION(substring IN string)`

Comment (GumpacG) [reply docs/user/ppl/functions/string.rst:161]: Should this just have `Argument type: STRING, STRING` then? Instead of `Argument type: STRING, STRING, INTEGER`

Comment (margarit-h) [reply docs/user/ppl/functions/string.rst]: Addressed, thanks!

**Review by GumpacG — APPROVED** [review]

File: docs/user/ppl/functions/string.rst [hunk 3]
```diff
@@ -150,6 +150,31 @@ Example::
     +---------------------+---------------------+
 
 
+POSITION
+------
+
+Description
+>>>>>>>>>>>
+
+Usage: The syntax POSITION(substr IN str) returns the position of the first occurrence of substring substr in string str. Returns 0 if substr is not in str. Returns NULL if any argument is NULL.
```
Comment (acarbonetto) [inline docs/user/ppl/functions/string.rst:159]:
```suggestion
Usage: The syntax `POSITION(substr IN str)` returns the position of the first occurrence of substring `substr` in string `str`. Returns `0` if `substr` is not in `str`. Returns `NULL` if any argument is `NULL`.
```

**Review by GabeFernandez310 — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]
